1-JavaScript/Week3/homework/ex4-shoppingCart.js

```diff
@@ -23,10 +23,21 @@ you have more than 3 items in your shopping cart the first item gets taken out.
 const shoppingCart = ['bananas', 'milk'];
 
 // ! Function to be tested
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
-}
+function addToShoppingCart(item) {
+  if(item===undefined){
+    return `You bought ${shoppingCart.join(', ')}!`;
+  }else{
+    if(shoppingCart.length<3){
+      shoppingCart.push(item);
+      return `You bought ${shoppingCart.join(', ')}!`;
+    }else{
+      shoppingCart.push(item);
+      shoppingCart.shift();
+      return `You bought ${shoppingCart.join(', ')}!`;
+    }
 
+  } 
+}
```

Comment:
You can make the function cleaner like this

```suggestion
function addToShoppingCart(item) {
    if (item && shoppingCart.length < 3){
      shoppingCart.push(item);
    } else if (item && shoppingCart.length ==> 3 ) {
      shoppingCart.push(item);
      shoppingCart.shift();
    }
   
     return `You bought ${shoppingCart.join(', ')}!`;
  } 
}
```

Reply:
Your solution is legible and logical. I updated mine, also!